ui/src/app/components/gke-app-configuration-panels/create-gke-app.spec.tsx

```diff
@@ -104,49 +116,231 @@ describe(CreateGkeApp.name, () => {
+
+      await component(appType, {
+        userApps: [
+          {
+            ...autodeleteTestApp,
+            dateAccessed: dateAccessed.toISOString(),
+            autodeleteThreshold,
+          },
+        ],
+      });
+
+      expect(
+        screen.queryByLabelText('Autodelete remaining days')
+      ).toHaveTextContent('5 days remain until deletion');
+    });
+
+    it('should show correct message when autodeleteRemainingDays is 0', async () => {
+      const now = new Date();
+      now.setDate(now.getDate() - 2); // Subtract 2 days
+      const dateAccessed = now;
+      const autodeleteThreshold = 2 * 24 * 60 + 1; // 2 days in minutes plus small buffer.
+
+      await component(appType, {
+        userApps: [
+          {
+            ...autodeleteTestApp,
+            dateAccessed: dateAccessed.toISOString(),
+            autodeleteThreshold,
+          },
+        ],
       });
 
-      it('should use the existing PD when creating', async () => {
-        const disk = stubDisk();
-        await component(appType, {
-          app: undefined,
-          disk,
+      // less than 1 day (0 days) remaining until deletion.
+      expect(
+        screen.queryByLabelText('Autodelete remaining days')
+      ).toHaveTextContent('App will be deleted within 1 day');
+    });
+
+    it('should not show autodeleteRemainingDays when app is not running', async () => {
+      // Render the component with the app not running
+      await component(appType);
+
+      // Query for the autodeleteRemainingDays text
+      const autodeleteRemainingDaysText = screen.queryByLabelText(
+        'Autodelete remaining days'
+      );
+
+      // Assert that the autodeleteRemainingDays text is not in the document
+      expect(autodeleteRemainingDaysText).not.toBeInTheDocument();
+    });
+
+    it.each([
+      ['there are no running apps', {}],
+      [
+        'the current app is DELETED',
+        {
+          userApps: [{ ...listAppsResponse(), status: AppStatus.DELETED }],
+        },
+      ],
+    ])(
+      `should allow machine type configuration when %s`,
+      async (_, propOverrides: Partial<CreateGkeAppProps>) => {
+        serverConfigStore.set({
+          config: {
+            ...serverConfigStore.get().config,
+            enableGKEAppMachineTypeChoice: true,
+          },
         });
 
+        const { container } = await component(appType, propOverrides);
+
+        // default is 4 CPUs, 15 GB RAM, n1-standard-4
+        const differentCpuCount = '2';
+        const differentRamCount = '7.5';
+        const differentMachineType = 'n1-standard-2';
+
+        // sanity check: does not match default
+        expect(differentMachineType).not.toEqual(
+          listAppsResponse().kubernetesRuntimeConfig.machineType
+        );
+
+        const cpuId = `${appTypeToString[appType]}-cpu`;
+        const cpuDropdown = expectDropdown(container, cpuId);
```

Comment:
`expectDropdown()` and `getDropdownOption()` are not very in the spirit of RTL, but I don't know of a better solution for PrimeReact Dropdown

Reply:
Going fwd, we should have a method in react-test-helper to test for dropdowns. Doing that, we then would have to make changes at one place rather than individual files once we user a terra dropdown

Reply:
these functions are in react-test-helpers.